src/NuGet.Core/NuGet.Build.Tasks/BuildTasksUtility.cs

```diff
@@ -550,7 +550,7 @@ private static async Task<RestoreSummary> PerformNuGetV2RestoreAsync(Common.ILog
             {
                 cacheContext.NoCache = noCache;
 
-                var downloadContext = new PackageDownloadContext(cacheContext, repositoryPath, directDownload: false)
+                var downloadContext = new PackageDownloadContext(cacheContext, repositoryPath, directDownload: false, settings)
```

Comment:
How about passing the the namespace configuration to minimize the exposure, instead of the complete ISettings object.

Reply:
Ok. Done.